webapp/src/main.rs

```diff
@@ -94,54 +91,51 @@ fn rocket() -> _
         // Submission endpoint
         .mount("/", routes![submit::post_submit])
         // Server states
-        .manage(backend_config)
+        .manage(RwLock::new(backend_config))
         .manage(sessions_tracker)
         .manage(logger)
         // Templaiting fairing
         .attach(rocket_dyn_templates::Template::fairing())
         // Sessions cleaner thread startup
         .attach(AdHoc::on_liftoff("Sessions cleaner", |rocket| Box::pin(async move 
             {
-                let tracker = rocket.state::<Arc<Mutex<SessionsTracker>>>()
+                let tracker = rocket.state::<Arc<SessionsTracker>>()
                     .unwrap().to_owned();
                 let logger = rocket.state::<Arc<slog::Logger>>().unwrap().to_owned();
-                let interval = rocket.state::<BackendConfig>()
-                    .unwrap().sessions_cleanup_interval;
+                let interval = rocket.state::<RwLock<BackendConfig>>()
+                    .unwrap().read().unwrap().sessions_cleanup_interval;
                 info!(logger, "Sessions cleaner started");
                 
                 thread::spawn(move ||
                 {
                     loop
                     {
-                        sleep(std::time::Duration::from_millis(interval));
-                        let mut locked = tracker.lock()
-                            .unwrap_or_else(|_| std::process::exit(1));
-                        let deleted = locked.delete_old();
-                        drop(locked);
+                        sleep(std::time::Duration::from_millis(interval));                           
+                        let deleted = tracker.delete_old();
                         info!(logger, "Deleted {} old sessions", deleted);
                     }
                 });
             })))
         // Sessions saver thread startup
         .attach(AdHoc::on_liftoff("Sessions saver", |rocket| Box::pin(async move 
             {
-                let tracker = rocket.state::<Arc<Mutex<SessionsTracker>>>()
+                let tracker = rocket.state::<Arc<SessionsTracker>>()
                     .unwrap().to_owned();
                 let logger = rocket.state::<Arc<slog::Logger>>().unwrap().to_owned();
-                let config = rocket.state::<BackendConfig>().unwrap();
+                let config = rocket.state::<RwLock<BackendConfig>>().unwrap()
+                    .read().unwrap();
                 let interval = config.sessions_save_interval;
                 let save_path = config.sessions_data_file.clone();
+
+                drop(config);
```

Comment:
won't the config be dropped automatically right after line 128?

Reply:
I don't see why it would be dropped automatically exactly right after line 128, but I get the point. It would only live for 2 more operations and this extra locking time could be disregarded, I mainly did it manually to explicitly show that we wouldn't need it anymore after that point.
And also I thought that this miniscule extra locking time could get some perfectionist angry.